packages/traceloop-sdk/tests/test_vertexai_instrumentation.py

```diff
@@ -0,0 +1,252 @@
+    )
+
+
+def test_vertexai_predict_async(exporter):
+    @aworkflow("predict_async")
+    async def async_predict_text() -> str:
+        """Ideation example with a Large Language Model"""
+
+        parameters = {
+            "max_output_tokens": 256,
+            "top_p": 0.8,
+            "top_k": 40,
+        }
+
+        model = TextGenerationModel.from_pretrained("text-bison@001")
+        response = await model.predict_async(
+            "Give me ten interview questions for the role of program manager.",
+            **parameters,
+        )
+
+        return response.text
+    asyncio.run(async_predict_text())
+
+    spans = exporter.get_finished_spans()
+    assert [span.name for span in spans] == [
+        "vertexai.predict",
+        "predict_async.workflow",
+    ]
+
+    vertexai_span = spans[0]
+    assert (
+        "Give me ten interview questions for the role of program manager." in vertexai_span.attributes["llm.prompts.0.user"]
+    )
+
+
+def test_vertexai_stream(exporter):
+    @workflow("stream_prediction")
+    def streaming_prediction() -> str:
+        """Streaming Text Example with a Large Language Model"""
+
+        text_generation_model = TextGenerationModel.from_pretrained("text-bison")
+        parameters = {
+            "max_output_tokens": 256,
+            "top_p": 0.8,
+            "top_k": 40,
+        }
+        responses = text_generation_model.predict_streaming(
+            prompt="Give me ten interview questions for the role of program manager.",
+            **parameters)
+
+        result = [response for response in responses]
+        return result
+    streaming_prediction()
+    spans = exporter.get_finished_spans()
+    assert [span.name for span in spans] == [
+        "vertexai.predict",
+        "stream_prediction.workflow",
+    ]
+
+    vertexai_span = spans[0]
+    assert (
+        "Give me ten interview questions for the role of program manager." in vertexai_span.attributes["llm.prompts.0.user"]
+    )
+
+
+def test_vertexai_stream_async(exporter):
+    @aworkflow("stream_prediction_async")
+    async def async_streaming_prediction() -> str:
+        """Streaming Text Example with a Large Language Model"""
+
+        text_generation_model = TextGenerationModel.from_pretrained("text-bison")
+        parameters = {
+            "max_output_tokens": 256,
+            "top_p": 0.8,
+            "top_k": 40,
+        }
+
+        responses = text_generation_model.predict_streaming_async(
+            prompt="Give me ten interview questions for the role of program manager.",
+            **parameters
+        )
+        result = [response async for response in responses]
+        return result
+    asyncio.run(async_streaming_prediction())
+
+    spans = exporter.get_finished_spans()
+    assert [span.name for span in spans] == [
+        "vertexai.predict",
+        "stream_prediction_async.workflow",
+    ]
+
+    vertexai_span = spans[0]
+    assert (
```

Comment:
Please assert all attributes, not just the prompt

Reply:
Done